mantidproject/mantid
# fix_file_extension property in inelastic reduction

**Description of work.**
Addresses #24058

An additional property fix_file_extension added to direct inelastic reduction scripts. If set to True, the reduction selects only files with specified file extension, avoiding issue when the file with different extension are selected when the reduction is run automatically. 

**Report to: Ross Steward** . 

**To test:**

This is small change shouild be easy to test. The unit test verifies the things its doing. 
Only files with the extension specified as input will be selected. 

Verify the logic of changes -- they are all tested and pretty minor




*This does not require release notes* because is minor, copied to Mantid on instrument as soon as unit tests are passing and the person requested it have been notified. 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
